experimental/overhead_matching/swag/evaluation/evaluate_swag.py

```diff
@@ -183,6 +184,29 @@ def get_motion_deltas_from_path(vigor_dataset: vd.VigorDataset, path: list[int])
     return motion_deltas
 
 
+def build_patch_index_from_particle(dataset: vd.VigorDataset, device: torch.device):
+    sat_patch_positions = dataset.get_patch_positions().to(device)
+    sat_patch_kdtree = build_kd_tree(sat_patch_positions)
+    num_patches = sat_patch_positions.shape[0]
+
+    def __inner__(particles: torch.Tensor):
+        # Search for up to 5 nearest neighbors.
+        MAX_DIST_DEG = 5e-4
```

Comment:
replace with new data in WAG config to convert satellite patch width/height and zoom level into a change in latitude/longitude